airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -615,40 +615,73 @@ public List<StandardSync> listStandardSyncs() throws ConfigNotFoundException, IO
 
   public List<StandardSync> listStandardSyncsUsingOperation(final UUID operationId)
       throws IOException {
-    final Result<Record> result = database.query(ctx -> ctx.select(CONNECTION.asterisk())
+    final Result<Record> connectionResult = database.query(ctx -> ctx.select(CONNECTION.asterisk(), CONNECTION_OPERATION.asterisk())
         .from(CONNECTION)
         .join(CONNECTION_OPERATION)
         .on(CONNECTION_OPERATION.CONNECTION_ID.eq(CONNECTION.ID))
         .where(CONNECTION_OPERATION.OPERATION_ID.eq(operationId))).fetch();
-    return getStandardSyncsFromResult(result);
+
+    final List<UUID> connectionIds = connectionResult.stream().map(r -> r.get(CONNECTION.ID)).toList();
+
+    final Result<Record> connectionOperationResult = database.query(ctx -> ctx.select(CONNECTION_OPERATION.asterisk()))
```

Comment:
maybe misunderstanding here - why do we need a second query if we are retrieving all the connection operations via the `CONNECTION_OPERATION.asterick` in the first query a few lines up?

Reply:
Reworked this to only need a single query